src/Symfony/Component/HttpClient/Internal/CurlClientState.php

```diff
@@ -52,8 +52,8 @@ public function __construct(int $maxHostConnections, int $maxPendingPushes)
         if (\defined('CURLMOPT_MAX_HOST_CONNECTIONS')) {
             $maxHostConnections = curl_multi_setopt($this->handle, \CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : \PHP_INT_MAX) ? 0 : $maxHostConnections;
         }
-        if (\defined('CURLMOPT_MAXCONNECTS') && 0 < $maxHostConnections) {
-            curl_multi_setopt($this->handle, \CURLMOPT_MAXCONNECTS, $maxHostConnections);
+        if (\defined('CURLMOPT_MAXCONNECTS') && null !== $maxConnections ??= 0 < $maxHostConnections ? $maxHostConnections : null) {
+            curl_multi_setopt($this->handle, \CURLMOPT_MAXCONNECTS, $maxConnections);
```

Comment:
instead of adding a new option, can you try removing this line and see if curl doesn't do better by default?

Reply:
OK, I re-read your description, you tell about it already.

Then what about setting $maxHostConnections to 500?

Reply:
@nicolas-grekas Unfortunately, setting `$maxHostConnections` only sets `CURLMOPT_MAX_HOST_CONNECTIONS`. This doesn't change `CURLMOPT_MAXCONNECTS` which by default is  4 times the number of added handlers. I've already tried this, and it didn't work.